openssl.go

```diff
@@ -138,6 +139,34 @@ func FIPS() bool {
 	}
 }
 
+// FIPSProvider returns true if the provider used by the default matches the `fips=yes` query.
+// Note that this function can return true even if [FIPS] returns false, because [FIPS] checks
+// whether the default properties contain `fips=1`. It will always return true for OpenSSL 3 if
+// [FIPS] returns true.
+// When using OpenSSL 1, this function always returns the same value as [FIPS].
+func FIPSProvider() bool {
+	if vMajor == 1 {
+		return FIPS()
+	}
+	provFn := func(props *C.char) C.GO_OSSL_PROVIDER_PTR {
+		md := C.go_openssl_EVP_MD_fetch(nil, algorithmSHA256, props)
+		if md == nil {
+			C.go_openssl_ERR_clear_error()
+			return nil
+		}
+		C.go_openssl_EVP_MD_free(md)
+		return C.go_openssl_EVP_MD_get0_provider(md)
+	}
+	// Load the provider with and without the `fips=yes` query.
```

Comment:
Comment above says fips=yes and comment in function mentions fips=1, would be good to update this to fips=1 in everywhere for consistency

Reply:
Good catch. THe correct value is `fips=yes`, so I'll use it in the function comment.